lib/bolt/config.rb

```diff
@@ -65,6 +66,15 @@ def initialize(**kwargs)
       end
     end
 
+    def deep_clone
```

Comment:
this should work for now but it may cause a bug in the future by not cloning arrays or any hashes we add later. It may be better to have this be more generic if it's not too expensive

Reply:
Sounds reasonable. I'm not quite sure what a generic implementation will look like yet...

Reply:
Many of the deep_* implementations rely on monkey-patching Object, such as https://github.com/rubyworks/facets/blob/27d9b7f5b3a507da17c0f48c1a7a95079359470d/lib/core/facets/kernel/deep_clone.rb. https://github.com/rubyworks/facets/blob/27d9b7f5b3a507da17c0f48c1a7a95079359470d/work/rejected/kernel/deep_clone.rb is an interesting option that might not require monkey-patching.

Reply:
Added a general deep_clone implementation.